src/gui/SubWindow.cpp

```diff
@@ -26,42 +26,245 @@
 
 #include "SubWindow.h"
 
+#include <QMdiArea>
 #include <QMoveEvent>
 #include <QResizeEvent>
 #include <QWidget>
 
+#include "embed.h"
+#include <QDebug>
+
 
 SubWindow::SubWindow(QWidget *parent, Qt::WindowFlags windowFlags)
   : QMdiSubWindow(parent, windowFlags)
 {
 	// initialize the tracked geometry to whatever Qt thinks the normal geometry currently is.
 	// this should always work, since QMdiSubWindows will not start as maximized
-	m_trackedNormalGeom = normalGeometry();
+  	m_trackedNormalGeom = normalGeometry();
+
+    // inits the colors
+    m_activeColor = Qt::SolidPattern;
+    m_textShadowColor = Qt::black;
+  	m_borderColor = Qt::black;
+
+    //close, minimize, maximize and restore(after minimize) buttons
+    m_closeBtn = new QPushButton( embed::getIconPixmap( "close" ), QString::null, this );
+    m_closeBtn->resize(17, 17);
```

Comment:
You should define a `const int btnSize = 17;` or something similar, and just easily replace every instance of `17` with that. Usually you want to use as low amount of 'magic numbers' as possible.